File: ecommerce/src/services/user/announcementBarService.js
import apiClient from "@/services/apiClient";

export const announcementBarService = {
  async getActive() {
    try {
      return await apiClient.request("/announcement-bar/active");
    } catch (error) {
      console.error("Không lấy được thông báo chạy:", error);
      return null;
    }
  },
};

export default announcementBarService;

File: ecommerce-backend/src/main/java/com/ecommerce/modules/announcement/service/AnnouncementBarService.java
package com.ecommerce.modules.announcement.service;

import com.ecommerce.modules.announcement.dto.request.AnnouncementBarRequest;
import com.ecommerce.modules.announcement.dto.response.AnnouncementBarResponse;

import java.util.List;

public interface AnnouncementBarService {
    AnnouncementBarResponse getActive();
    List<AnnouncementBarResponse> getAdminList();
    AnnouncementBarResponse create(AnnouncementBarRequest request);
    AnnouncementBarResponse update(Long id, AnnouncementBarRequest request);
    AnnouncementBarResponse toggleActive(Long id);
    void delete(Long id);
}
